### PR TITLE
Have the elapsed time affecting the amount of movement and rotation.

### PR DESCRIPTION
**Pull request by negin-z**
- Capture the elapsed time between two consecutive rendered frames.
- Incorporate the elapsed time in calculating the movement distance and rotation.

### CODE REVIEW

File: sdl_viewer/src/camera.rs [hunk 1]
```diff
@@ -152,7 +153,9 @@ impl Camera {
             self.pan += pan.normalize();
         }
 
-        const TURNING_SPEED : Rad<f32> = Rad(0.015);
+        let elapsed_seconds = elapsed.num_milliseconds() as f32 / 1000.;
```
Comment (thomasschiwietz) [inline sdl_viewer/src/camera.rs]: Replace by num_seconds()

File: sdl_viewer/src/camera.rs [hunk 2]
```diff
@@ -207,13 +210,13 @@ impl Camera {
     }
 
     pub fn pan(&mut self, x: f32, y: f32, z: f32) {
```
Comment (thomasschiwietz) [inline sdl_viewer/src/camera.rs]: There are duplicated functions for rotation and panning:
-  mouse_drag_pan() and pan()
-  mouse_drag_rotate() and rotate(). 

The mouse_* functions use constants (100) inside of the function while the non-mouse functions use constants outside of the function. Could we unify this to two function such as rotate(up, around) and pan(x,y,z) and set all  constants from the call site? Specifically, I find it inconsistent that Joystick computes the constants at the call site while mouse does not.

File: sdl_viewer/src/camera.rs (hunk 1 above)
Comment (negin-z) [reply sdl_viewer/src/camera.rs]: done!

File: sdl_viewer/src/camera.rs (hunk 2 above)
Comment (negin-z) [reply sdl_viewer/src/camera.rs]: Created a ticket to follow-up on it as a clean-up task

**Comment by negin-z** [comment]
@thomasschiwietz We need to merge this final commit. It reverts a code clean-up which breaks the navigation in sdl-viewer.

**Comment by negin-z** [comment]
I signed it!

**Comment by SirVer** [comment]
@wally-the-cartographer merge

**Comment by wally-the-cartographer** [comment]
Merge requested by authorized user SirVer. Merge queue now has a length of 1.

**Pull request merged**